src/components/GenerativeDirectAnswerDisplay.tsx

```diff
@@ -0,0 +1,137 @@
+import { 
+  GenerativeDirectAnswerResponse, 
+  useSearchActions, 
+  useSearchState, 
+  SearchTypeEnum, 
+  Result 
+} from '@yext/search-headless-react';
+import { useComposedCssClasses } from '../hooks';
+import { executeGenerativeDirectAnswer } from '../utils/search-operations';
+import ReactMarkdown from "react-markdown";
+import React from 'react';
+
+/**
+ * The CSS class interface used for {@link GenerativeDirectAnswerDisplay}.
+ *
+ * @public
+ */
+export interface GenerativeDirectAnswerDisplayCssClasses {
+  generativeDirectAnswerTitle?: string,
+  generativeDirectAnswerText: string,
+  generativeDirectAnswerDivider: string,
+  generativeDirectAnswerContainer?: string,
+  generativeDirectAnswerCitations: string,
+  generativeDirectAnswerCitation?: string,
+  generativeDirectAnswerCitationTitle?: string,
+  generativeDirectAnswerCitationSnippet?: string,
+}
+
+const builtInCssClasses: Readonly<GenerativeDirectAnswerDisplayCssClasses> = {
+  generativeDirectAnswerTitle: 'text-xl',
+  generativeDirectAnswerText: 'mt-4',
+  generativeDirectAnswerDivider: 'border-b border-gray-200 w-full pb-6 mb-6',
+  generativeDirectAnswerContainer: 'p-6 border border-gray-200 rounded-lg shadow-sm',
+  generativeDirectAnswerCitations: 'mt-4 flex overflow-x-auto gap-4',
+  generativeDirectAnswerCitation: 'p-4 border border-gray-200 rounded-lg shadow-sm bg-slate-100 flex flex-col basis-64 text-sm text-neutral',
+  generativeDirectAnswerCitationTitle: 'font-bold',
+  generativeDirectAnswerCitationSnippet: 'line-clamp-2 text-ellipsis break-words',
+};
+
+/**
+ * Props for {@link GenerativeDirectAnswerDisplay}.
+ *
+ * @public
+ */
+export interface GenerativeDirectAnswerDisplayProps<T> {
+  /** CSS classes for customizing the component styling. */
+  customCssClasses?: GenerativeDirectAnswerDisplayCssClasses
+}
+
+/**
+ * Displays the AI generated answer of a generative direct answer.
+ *
+ * @public
+ *
+ * @param props - {@link GenerativeDirectAnswerDisplayProps}
+ * @returns A React element for the generative direct answer display, or null if there is no generated answer
+ */
+export function GenerativeDirectAnswerDisplay<T>({
```

Comment:
I believe most, if not all, BlahDisplay components are meant to be internal. the public exported component would just be GenerativeDirectAnswer. see index.ts